src/main/java/Model/Utilities/Check.java

```diff
@@ -0,0 +1,55 @@
+package Model.Utilities;
+
+import Model.Game;
+import Model.Position;
+import Model.Pieces.Piece;
+import Model.Pieces.PieceType;
+
+import java.util.ArrayList;
+
+import static java.util.Objects.isNull;
+
+public class Check {
+
+
+    /**
+     *
+     * @param game
+     * @param whitesTurn
+     * @return
+     */
+    public static boolean isTheKingUnderAttack(Game game, boolean whitesTurn){
+        ArrayList<Piece> attackingPieces = game.getClonedColouredPieces(!whitesTurn);
+        Piece kingPiece = game.getClonedColouredPieces(whitesTurn).stream().filter(piece3 -> piece3.getPieceType() == PieceType.King).findFirst().orElse(null);
+
+        if (isNull(kingPiece)){
+            return false;
+        }
+        Position kingPosition = kingPiece.getPosition();
+        for (Piece piece:attackingPieces) {
+            ArrayList<Position> possibleMoves = Moves.getAllMovesIncludingIllegalMoves(piece.getPosition(), game);
+            if (possibleMoves.stream().anyMatch(position -> position.getFile()==kingPosition.getFile() && position.getRank()==kingPosition.getRank())){
+                return true;
+            }
+        }
+        return false;
+    }
+
+
+    /**
+     *
+     * @param game
+     * @param whitesTurn
+     * @return
+     */
+    public static boolean isThereALegalMove(Game game, boolean whitesTurn){
+        ArrayList<Piece> defendingPieces = game.getClonedColouredPieces(whitesTurn);
```

Comment:
also why cloning here? this cant be done on the original list? are you are scared of exposing the original list so that it wont be changed?

Reply:
Yeah I just think that only Game should be dealing with the actual objects, and that everything else should be using a clone for safety. Maybe for Check not necessary but is necessary in other situations